fluentd/configs.d/filter/retag_journal.conf

```diff
@@ -0,0 +1,66 @@
+# "io.kubernetes.container.name": "registry",
+# "io.kubernetes.pod.name": "docker-registry-1-hhoj0",
+# "io.kubernetes.pod.namespace": "default",
+# "io.kubernetes.pod.terminationGracePeriod": "30",
+# "io.kubernetes.pod.uid": "ae3a9bdc-1f66-11e6-80a2-fa163e2fff3a",
+# docker ps:
+# da531d7cea12 172.30.38.242:5000/logging/logging-elasticsearch@sha256:4888a2a3f875555a46d233f9856964a8eaac716259599db0062051ba5647187e   "sh /opt/app-root/src"   About an hour ago   Up About an hour
+# k8s_elasticsearch.8d3af8fd_logging-es-ops-gkfv06wi-1-7y682_logging_9612f891-1f68-11e6-80a2-fa163e2fff3a_d1cc32e1
+# oc get pods -o yaml:
+#      kubernetes.io/created-by: |
+# {"name":"logging-es-ops-gkfv06wi-1","uid":"92d90581-1f68-11e6-80a2-fa163e2fff3a","apiVersion":"v1","resourceVersion":"829"}}
+# openshift.io/deployment-config.name: logging-es-ops-gkfv06wi
+# openshift.io/deployment.name: logging-es-ops-gkfv06wi-1
+# uid: 9612f891-1f68-11e6-80a2-fa163e2fff3a
+# image: 172.30.38.242:5000/logging/logging-elasticsearch@sha256:4888a2a3f875555a46d233f9856964a8eaac716259599db0062051ba5647187e
+# containerStatuses:
+#  - containerID: docker://da531d7cea12bf714f90c26d0fb9f275b02321d1bccd10013f087be33f4e6e5e
+#    image: 172.30.38.242:5000/logging/logging-elasticsearch@sha256:4888a2a3f875555a46d233f9856964a8eaac716259599db0062051ba5647187e
+#    imageID: docker://sha256:d5a02bf63b1f3e0d1aa06bf28928e181f67ab2c80b2a1d1fd1ba310f11c12955
+# another example:
+# CONTAINER_NAME=k8s_bob.94e110c7_bob-iq0d4_default_2d67916a-1eac-11e6-94ba-001c42e13e5d_8b4b7e3d
+# From this, we can extract:
+#    container name in pod: bob
+#    pod name: bob-iq0d4
+#    namespace: default
+#    pod uid: 2d67916a-1eac-11e6-94ba-001c42e13e5d
+
+# in addition, there are a few other differences about /var/log/messages and journal messages
+# * the default rsyslog.conf on el7 will only log messages to /var/log/messages at INFO level
+#   or higher
+# * the default rsyslog.d includes 21-cloudinit.conf which redirects all cloud-init messages
+#   to /var/log/cloud-init.log
+
+@include cloud_init.conf
+
+@include max_priority.conf
+
+<match journal>
+  @type rewrite_tag_filter
+  # see if this is a kibana container for special log handling
+  rewriterule1 CONTAINER_NAME ^k8s_.*kibana          kubernetes.journal.container.kibana
+  # mark default container logs as system logs
+  rewriterule2 CONTAINER_NAME ^k8s_.*default         kubernetes.journal.container._default_
+  # mark openshift-infra container logs as system logs
+  rewriterule3 CONTAINER_NAME ^k8s_.*openshift-infra kubernetes.journal.container._openshift-infra_
+  # mark openshift container logs as system logs
+  rewriterule4 CONTAINER_NAME ^k8s_.*openshift       kubernetes.journal.container._openshift_
+  # not kibana - see if this is a kubernetes container log
+  rewriterule5 CONTAINER_NAME ^k8s_.*fluentd         kubernetes.journal.container.fluentd
+  rewriterule6 CONTAINER_NAME ^k8s_                  kubernetes.journal.container
```

Comment:
are we losing the project that the logs came from in this case, or do we recover that with the metadata later on?